src/tools/ontoweave.py

```diff
@@ -0,0 +1,107 @@
+
+    do.add_argument("-s", "--schema", metavar="FILE", default="schema.yaml",
+        help="The BioCypher schema file (the one managing node and edge types). [default: %(default)s]")
+
+    do.add_argument("-p", "--parallel", metavar="NB_CORES", default="0",
+        help=f"Number of processor cores to use when processing with multi-threading. `0` means a sequantial processing (no parallelization, the default). Use 'auto' to let {name} do its best to use a good number. [default: %(default)s]")
+
+    do.add_argument("-i", "--import-script-run", action="store_true",
+        help=f"If passed {name} will call the import scripts created by Biocypher for you.")
+
+    do.add_argument("-S", "--prop-sep", metavar="CHARACTER", default = ";",
+        help="The character used to separate property values fusionned in the same property at the reconciliation step. [default: %(default)s]")
+
+    do.add_argument("-a", "--type-affix", choices=["suffix","prefix","none"], default="none",
+        help="Where to add the type string to the ID label. [default: %(default)s]")
+
+    do.add_argument("-A", "--type-affix-sep", metavar="CHARACTER", default=":",
+        help="Character used to separate the label from the type affix. [default: %(default)s]")
+
+    do.add_argument("-l", "--log-level", default="WARNING",
+        choices=['DEBUG', 'INFO', 'WARNING', 'ERROR', 'CRITICAL'],
+        help="Configure the log level. [default: %(default)s]")
+
+    asked = do.parse_args()
+
+    logging.basicConfig(level = asked.log_level)
+
+    logging.info("OntoWeave parameters:")
+
+    logging.info(f"\tconfig: `{asked.config}`")
+    logging.info(f"\tschema: `{asked.schema}`")
+    logging.info(f"\tprop-sep: `{asked.prop_sep}`")
+    logging.info(f"\ttype-affix: `{asked.type_affix}`")
+    logging.info(f"\ttype-affix-sep: `{asked.type_affix_sep}`")
+    logging.info(f"\timport-script-run: `{asked.import_script_run}`")
+    logging.info(f"\tlog-level: `{asked.log_level}`")
+
+    logging.info(f"\tasked mappings: `{asked.mapping}`")
+    asked_mapping = []
+    if asked.mapping == ["STDIN"]:
+        while True:
+            try:
+                item = sys.stdin.readline()
+            except UnicodeDecodeError:
+                continue
+            except KeyboardInterrupt:
+                break
+            if not item:
+                break
+            asked_mapping.append(item)
+    else:
+        asked_mapping = asked.mapping
+
+    logging.info(f"\tparsed mappings: {asked_mapping}")
+    mappings = {}
+    for data_map in asked_mapping:
+        if ":" not in data_map:
+            msg = f"Cannot parse the DATA:MAPPING `{data_map}`, I cannot find the colon character."
+            logging.error(msg)
+            raise RuntimeError(msg)
+        data,map = data_map.split(":")
+        mappings[data] = map
+        logging.info(f"\t\t`{data}` => `{map}`")
+
+    if asked.parallel == "auto":
+        parallel = min(32, (os.process_cpu_count() or 1) + 4)
```

Comment:
Good catch. This was added in Python 3.13, and we require Python 3.9 (but I ran on local install).
It is however a better option than `cpu_count`, since [it takes into account either the processors affinity, or a global config set on the Python interpretor](https://docs.python.org/3/library/os.html#os.process_cpu_count). I am hesitating to require Python 3.13. What do you think?

Reply:
Good idea, I tried changing the poetry environments locally and Python 3.13 seems to work.